src/main/java/maksym/fedorenko/bookstore/model/Order.java

```diff
@@ -0,0 +1,72 @@
+package maksym.fedorenko.bookstore.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Table(name = "orders")
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted=false")
+@Data
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "user_id")
```

Comment:
just wondering

`@JoinColumn(nullable = false)` == `@AnyToOne(optional = false)` ? :)

I saw that you did some mapping with both `nullable` and `optional`, but here you specified just `optional`

try to find out whether these both are equal

Reply:
From stackoverflow:
"optional = false is only for checking this constraint in runtime. nullable = false creates database constraint. For applications, to also set optional = false makes sense, because it is evaluated faster than going to database and check that constraint there."
Fail-Fast approach, Nazar, I think you know 🙃